packages/libraries/client/src/yoga.ts

```diff
@@ -0,0 +1,61 @@
+import { DocumentNode, GraphQLSchema, parse } from 'graphql';
+import type { Plugin } from 'graphql-yoga';
+import LRU from 'tiny-lru';
+import { createHive } from './client.js';
+import type { CollectUsageCallback, HiveClient, HivePluginOptions } from './internal/types.js';
+import { isHiveClient } from './internal/utils.js';
+
+export function useHive(clientOrOptions: HiveClient): Plugin;
+export function useHive(clientOrOptions: HivePluginOptions): Plugin;
+export function useHive(clientOrOptions: HiveClient | HivePluginOptions): Plugin {
+  const hive = isHiveClient(clientOrOptions)
+    ? clientOrOptions
+    : createHive({
+        ...clientOrOptions,
+        agent: {
+          name: 'hive-client-yoga',
+          ...clientOrOptions.agent,
+        },
+      });
+
+  void hive.info();
+
+  const lru = LRU<DocumentNode>(10_000);
```

Comment:
Not a fan of hardcoding this here. If we could somehow share the parser cache from yoga that would be cool. Any idea on whether that is possible @ardatan?